modules/charon-core/src/main/java/org/wso2/charon3/core/schema/AbstractValidator.java

```diff
@@ -143,6 +197,92 @@ private static void validateSCIMObjectForRequiredSubAttributes(AbstractAttribute
         }
     }
 
+    /**
+     * Validation the patch operation effect on sub attributes of a scim attribute.
+     *
+     * @param oldAttribute    Old scim attribute.
+     * @param newAttribute    Updated scim attribute.
+     * @param attributeSchema Attribute schema of the attribute.
+     * @param scimObject      Scim object.
+     * @throws CharonException     When error occurred during the validation.
+     * @throws BadRequestException When error occurred due to the client issues.
+     */
+    private static void validatePatchOperationEffectForRequiredSubAttributes(AbstractAttribute oldAttribute,
+                                                                             AbstractAttribute newAttribute,
+                                                                             AttributeSchema attributeSchema,
+                                                                             AbstractSCIMObject scimObject)
+            throws CharonException, BadRequestException {
+
+        if (newAttribute == null || attributeSchema == null || oldAttribute == null) {
+            return;
+        }
+        List<AttributeSchema> subAttributesSchemaList = attributeSchema.getSubAttributeSchemas();
+        if (subAttributesSchemaList == null) {
+            return;
+        }
+        for (AttributeSchema subAttributeSchema : subAttributesSchemaList) {
+            // Nothing to validate id the attribute schema has required=false.
+            if (!subAttributeSchema.getRequired()) {
+                continue;
+            }
+            if (newAttribute instanceof ComplexAttribute) {
+                // If the sub attribute contained in the old attribute but not in the new attribute.
+                if (newAttribute.getSubAttribute(subAttributeSchema.getName()) == null
+                        && oldAttribute.getSubAttribute(subAttributeSchema.getName()) != null) {
+                    String error = "Required sub attribute: " + subAttributeSchema.getName()
+                            + " is missing in the SCIM Attribute: " + newAttribute.getName();
+                    throw new BadRequestException(error, ResponseCodeConstants.INVALID_VALUE);
+                } else if (newAttribute.getSubAttribute(
+                        subAttributeSchema.getName()) instanceof SimpleAttribute) {
```

Comment:
we might be using empty values for required attributes in admin force password reset flow, Please check whether this affects that flow.

Reply:
@ayshsandu